airflow/models.py

```diff
@@ -3389,7 +3390,14 @@ def create_dagrun(self,
             state=state
         )
         session.add(run)
-        session.commit()
+        try:
```

Comment:
In addition, please query the database to ask if a matching dag run already exists.